### ISSUE TITLE
FSC embeds two version attibutes when developer uses wild cards

### ISSUE THREAD

**Issue by KevinRansom** [dotnet/fsharp#12545]
Compile this using the F# compiler:
````
open System
open System.Reflection;
open System.Configuration.Assemblies

[<assembly:AssemblyVersion("1.0.*")>]
do ()
````

Yields this metadata: note two assemblyversion attributes.
````
// C:\Users\codec\AppData\Local\Temp\CompilerAssert\dlvv20i0.fyi\sfnztls3.exe
// sfnztls3, Version=1.0.8030.2219, Culture=neutral, PublicKeyToken=null
// Global type: <Module>
// Entry point: <StartupCode$sfnztls3>.$Sfnztls3.main@
// Architecture: AnyCPU (64-bit preferred)
// Runtime: v4.0.30319
// Hash algorithm: SHA1

using System.Reflection;
using Microsoft.FSharp.Core;

[assembly: FSharpInterfaceDataVersion(2, 0, 0)]
[assembly: AssemblyVersion("1.0.*")]
[assembly: AssemblyVersion("1.0.8030.2219")]
````

Expected attribute like a C# compile.
````

// C:\Users\codec\source\repos\ConsoleApp6\ConsoleApp6\bin\Debug\net48\ConsoleApp6.exe
// ConsoleApp6, Version=1.0.8030.30883, Culture=neutral, PublicKeyToken=null
// Global type: <Module>
// Entry point: Example.Main
// Architecture: AnyCPU (64-bit preferred)
// Runtime: v4.0.30319
// Hash algorithm: SHA1
// Debug info: Loaded from portable PDB: C:\Users\codec\source\repos\ConsoleApp6\ConsoleApp6\bin\Debug\net48\ConsoleApp6.pdb

using System.Diagnostics;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Runtime.Versioning;

[assembly: CompilationRelaxations(8)]
[assembly: RuntimeCompatibility(WrapNonExceptionThrows = true)]
[assembly: Debuggable(DebuggableAttribute.DebuggingModes.Default | DebuggableAttribute.DebuggingModes.DisableOptimizations | DebuggableAttribute.DebuggingModes.IgnoreSymbolStoreSequencePoints | DebuggableAttribute.DebuggingModes.EnableEditAndContinue)]
[assembly: TargetFramework(".NETFramework,Version=v4.8", FrameworkDisplayName = ".NET Framework 4.8")]
[assembly: AssemblyVersion("1.0.8030.30883")]
````

**Comment by KevinRansom** [comment]
Actualy it's not with wild cards it's every version.
````
// #Regression #Attributes #Assemblies 
// Verify ability to put wildcards in assembly version

open System
open System.Reflection;
open System.Configuration.Assemblies

[<assembly:AssemblyVersion("1.0.1.2")>]
do ()
````

````
// C:\Users\codec\AppData\Local\Temp\CompilerAssert\m4c5bog3.kxf\alszyl0t.exe
// alszyl0t, Version=1.0.1.2, Culture=neutral, PublicKeyToken=null
// Global type: <Module>
// Entry point: <StartupCode$alszyl0t>.$Gpkhgqoh.main@
// Architecture: AnyCPU (64-bit preferred)
// Runtime: v4.0.30319
// Hash algorithm: SHA1

using System.Reflection;
using Microsoft.FSharp.Core;

[assembly: FSharpInterfaceDataVersion(2, 0, 0)]
[assembly: AssemblyVersion("1.0.1.2")]
[assembly: AssemblyVersion("1.0.1.2")]
````

I imagine that Ilspy displays the in assembly attribute, along with the native attribute metadata.

**Comment by KevinRansom** [comment]
Interestingly, C# doesn't embed the il version of the assemblyversionatrribute.
this code:
````
using System;
using System.Reflection;
using System.Reflection.Emit;

[assembly: System.Reflection.AssemblyVersion("1.0.1.1")]
class Example
{
    public static void Main()
    {
        var attr = Assembly.GetExecutingAssembly().GetCustomAttributes(typeof(AssemblyVersionAttribute), true)[0] as AssemblyVersionAttribute;
        Console.WriteLine($"{attr.ToString()}");
    }
}
````

produces this output:
````
C:\Users\codec\source\repos\ConsoleApp6\ConsoleApp6>bin\Debug\net48\ConsoleApp6.exe

Unhandled Exception: System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Example.Main() in C:\Users\codec\source\repos\ConsoleApp6\ConsoleApp6\Program.cs:line 10
````

**Comment by KevinRansom** [comment]
I'm not sure what to do about this.  Compatiblity with C# is the most sensible, but that it is technically a breaking change, although reflection apps are not guaranteed to continue to work identically.

The existing behavior is slightly idiosyncratic, given that an app with a generated version number: E.g:  "1.0.* will have an assemblyversionattribute of: 1.0.* and a version of something like : "1.0.888.1234".
And when the version is set using: --version:1.2.3.4 no assemblyversionattribute is added to the assembly.

The correct way to get the version # is probably:
````
    let version = Assembly.GetExecutingAssembly().GetName().Version
````

**Comment by heronbpv** [comment]
Wouldn't stop declaring the attribute twice be enough in this case?

**Comment by KevinRansom** [comment]
fixed by: #12550
